Sources/XiEditor/EditView.swift

```diff
@@ -604,6 +613,22 @@ final class EditView: NSView, NSTextInputClient, TextPlaneDelegate {
 
                 let x = dataSource.gutterWidth - (gutterXPad + CGFloat(gutterTL.width))
                 let y0 = yOff + dataSource.textMetrics.ascent + linespace * CGFloat(lineIx)
+
+                for annotationType in gutterAnnotations {
+                    var annotations = annotationsForLines[lineIx]![annotationType]!
+                    let annotation = annotations.next()
+
+                    if annotation != nil {
+                        let color = annotationColor(for: annotation!.annotation)
+
+                        if annotationType == AnnotationType.deleted {
+                            renderer.drawSolidRect(x: GLfloat(dataSource.gutterWidth - CGFloat(gutterAnnotationWidth)), y: GLfloat(yOff + dataSource.textMetrics.ascent + linespace * CGFloat(lineIx - 1)), width: GLfloat(gutterAnnotationWidth), height: GLfloat(3.0), argb: color)
+                        } else {
+                            renderer.drawSolidRect(x: GLfloat(dataSource.gutterWidth - CGFloat(gutterAnnotationWidth)), y: GLfloat(yOff + linespace * CGFloat(lineIx)), width: GLfloat(gutterAnnotationWidth), height: GLfloat(linespace), argb: color)
```

Comment:
Nit: There should be newlines after each parameter name for these kinds of long function calls if they don't fit entirely on one line. I think it looks nicer and is generally easier to read.

So this could be 
```
 if annotationType == AnnotationType.deleted {
                            renderer.drawSolidRect(x: GLfloat(dataSource.gutterWidth - CGFloat(gutterAnnotationWidth)),
                                                   y: GLfloat(yOff + dataSource.textMetrics.ascent + linespace * CGFloat(lineIx - 1)),
                                                   width: GLfloat(gutterAnnotationWidth),
                                                   height: GLfloat(3.0),
                                                   argb: color)
                        } else {
                            renderer.drawSolidRect(x: GLfloat(dataSource.gutterWidth - CGFloat(gutterAnnotationWidth)),
                                                   y: GLfloat(yOff + linespace * CGFloat(lineIx)),
                                                   width: GLfloat(gutterAnnotationWidth),
                                                   height: GLfloat(linespace),
                                                   argb: color)
                        }
```